backend/icpy/agent/tools/imagen_tool.py

```
@@ -568,41 +610,100 @@ def _resize_image(
-                model = genai.GenerativeModel(model_name)  # type: ignore[attr-defined]
+                return response, None
+                
+            elif GENAI_PROVIDER == 'google-generativeai-legacy':
+                # Legacy SDK path (no native aspect_ratio support)
+                model = self._legacy_genai.GenerativeModel(model_name)  # type: ignore[attr-defined]
+                logger.warning("[ImagenTool] Using legacy SDK - aspect_ratio will be handled via post-resize")
                 return model.generate_content(content), None
+            else:
+                return None, f"No valid Google SDK provider available (got: {GENAI_PROVIDER})"
+                
         except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)